packages/vaadin-grid/src/vaadin-grid-selection-column.js

```diff
@@ -96,99 +77,41 @@ class GridSelectionColumnElement extends GridColumnElement {
       },
 
       /** @private */
-      _indeterminate: Boolean,
+      __indeterminate: Boolean,
 
       /**
        * The previous state of activeItem. When activeItem turns to `null`,
        * previousActiveItem will have an Object with just unselected activeItem
        * @private
        */
-      _previousActiveItem: Object,
+      __previousActiveItem: Object,
 
       /** @private */
-      _selectAllHidden: Boolean
+      __selectAllHidden: Boolean
     };
   }
 
   static get observers() {
-    return ['_onSelectAllChanged(selectAll)'];
-  }
-
-  /** @private */
-  _pathOrHeaderChanged(
-    path,
-    header,
-    headerCell,
-    footerCell,
-    cells,
-    renderer,
-    headerRenderer,
-    bodyTemplate,
-    headerTemplate
-  ) {
-    // As a special case, allow overriding the default header / body templates
-    if (cells.value && (path !== undefined || renderer !== undefined)) {
-      this._bodyTemplate = bodyTemplate = undefined;
-      this.__cleanCellsOfTemplateProperties(cells.value);
-    }
-    if (headerCell && (header !== undefined || headerRenderer !== undefined)) {
-      this._headerTemplate = headerTemplate = undefined;
-      this.__cleanCellsOfTemplateProperties([headerCell]);
-    }
-    super._pathOrHeaderChanged(
-      path,
-      header,
-      headerCell,
-      footerCell,
-      cells,
-      renderer,
-      headerRenderer,
-      bodyTemplate,
-      headerTemplate
-    );
-  }
-
-  /** @private */
-  __cleanCellsOfTemplateProperties(cells) {
-    cells.forEach((cell) => {
-      cell._content.innerHTML = '';
```

Comment:
Haven't covered in this PR yet. Probably, I need to reset the cell content somewhere in `__setColumnTemplateOrRenderer`:

https://github.com/vaadin/web-components/pull/1967/files#diff-37fc749a7a99c4f9c6690e4879c4657c493c59e185d41c3abdff95d08963c76dR340

```
if (cell._renderer !== renderer) {
  cell._content.innerHTML = '';
}
```